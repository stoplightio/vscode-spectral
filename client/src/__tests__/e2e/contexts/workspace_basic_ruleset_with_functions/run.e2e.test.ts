/* eslint-disable require-jsdoc */

import { expect } from 'chai';
import * as chaiJestSnapshot from 'chai-jest-snapshot';
import * as vscode from 'vscode';
import * as path from 'path';

import { openFile, activate } from '../../helper';
import { workspace } from 'vscode';

suiteSetup(async () => {
  chaiJestSnapshot.resetSnapshotRegistry();
  await activate();
});

setup(function() {
  // eslint-disable-next-line no-invalid-this
  chaiJestSnapshot.configureUsingMochaContext(this);
});

suite('Workspace, basic ruleset with functions', () => {
  suite('No diagnostics for empty files', () => {
    ['empty.yaml', 'empty.json'].forEach((fixture) => {
      test(`${fixture}`, async () => {
        const diags = await lint(['empty', fixture]);
        expect(diags).to.matchSnapshot();
      });
    });
  });

  suite('No diagnostics for valid files', () => {
    ['simple.yaml', 'simple.json'].forEach((fixture) => {
      test(`${fixture}`, async () => {
        const diags = await lint(['valid', fixture]);
        expect(diags).to.matchSnapshot();
      });
    });
  });

  suite('Invalid files trigger generation of diagnostics', () => {
    ['simple.yaml', 'simple.json'].forEach((fixture) => {
      test(`${fixture}`, async () => {
        const diags = await lint(['invalid', fixture]);

        expect(diags).to.matchSnapshot();
      });
    });
  });

  const lint = async (pathSegments: string[]) => {
    const docPath = path.resolve(workspace.rootPath as string, ...pathSegments);

    const docUri = vscode.Uri.file(docPath);
    await openFile(docUri);

    return vscode.languages.getDiagnostics(docUri);
  };
});
