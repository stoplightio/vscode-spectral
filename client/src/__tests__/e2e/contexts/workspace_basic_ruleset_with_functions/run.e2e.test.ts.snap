// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace, basic ruleset with functions Invalid files trigger generation of diagnostics simple.json 1`] = `
Array [
  Object {
    "code": "demand-newest-oas3",
    "message": "3.0.0 must match 3.1.0",
    "range": Array [
      Object {
        "character": 13,
        "line": 1,
      },
      Object {
        "character": 20,
        "line": 1,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
  Object {
    "code": "valid-document-version",
    "message": "1.0.0 must match 2.0.0",
    "range": Array [
      Object {
        "character": 15,
        "line": 3,
      },
      Object {
        "character": 22,
        "line": 3,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
]
`;

exports[`Workspace, basic ruleset with functions Invalid files trigger generation of diagnostics simple.yaml 1`] = `
Array [
  Object {
    "code": "demand-newest-oas3",
    "message": "3.0.0 must match 3.1.0",
    "range": Array [
      Object {
        "character": 9,
        "line": 0,
      },
      Object {
        "character": 14,
        "line": 0,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
  Object {
    "code": "valid-document-version",
    "message": "1.0.0 must match 2.0.0",
    "range": Array [
      Object {
        "character": 11,
        "line": 2,
      },
      Object {
        "character": 16,
        "line": 2,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
]
`;

exports[`Workspace, basic ruleset with functions No diagnostics for empty files empty.json 1`] = `Array []`;

exports[`Workspace, basic ruleset with functions No diagnostics for empty files empty.yaml 1`] = `Array []`;

exports[`Workspace, basic ruleset with functions No diagnostics for valid files simple.json 1`] = `Array []`;

exports[`Workspace, basic ruleset with functions No diagnostics for valid files simple.yaml 1`] = `Array []`;
